Bookish.Web/Models/CatalogueViewModel.cs

```diff
@@ -1,15 +1,34 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Bookish.DataAccess;
 
 namespace Bookish.Web.Models
 {
     public class CatalogueViewModel
     {
         public IEnumerable<CatalogueEntry> Catalogue { get; }
+        public string Filter { get; }
+        public int Page { get; }
+        public int LastPage { get; }
 
-        public CatalogueViewModel(IEnumerable<CatalogueEntry> catalogue)
+        public CatalogueViewModel(IEnumerable<CatalogueEntry> catalogue, int page = 1, string filter = "")
         {
-            Catalogue = catalogue;
+            Page = page;
+            Filter = filter;
+
+            Catalogue = catalogue.ToList();
+            LastPage = (int)Math.Ceiling(((decimal) Catalogue.Count() / 5));
```

Comment:
This has a few more brackets than it needs I think 🙂 Something like this should work:
```
LastPage = (int)Math.Ceiling(Catalogue.Count() / (double)5);
```
🙂